README.md

```diff
@@ -911,6 +911,150 @@ module.exports = {
 };
 ```
 
+### Multiple Themes
+
+**webpack.config.js**
+
+```js
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+module.exports = {
+  entry: "./src/index.js",
+  module: {
+    rules: [
+      {
+        test: /\.s[ac]ss$/i,
+        oneOf: [
+          {
+            resourceQuery: "?dark",
+            use: [
+              Self.loader,
+              "css-loader",
+              {
+                loader: "sass-loader",
+                options: {
+                  additionalData: `@use 'dark-theme/vars' as vars;`,
+                },
+              },
+            ],
+          },
+          {
+            use: [
+              Self.loader,
+              "css-loader",
+              {
+                loader: "sass-loader",
+                options: {
+                  additionalData: `@use 'light-theme/vars' as vars;`,
+                },
+              },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+  plugins: [
+    new Self({
+      filename: "[name].css",
+      attributes: {
+        id: "theme",
+      },
+    }),
+  ],
+};
+```
+
+**src/index.js**
+
+```js
+import "./style.scss";
+
+let theme = "light";
+const themes = {};
+
+themes[theme] = document.querySelector("#theme");
+
+async function loadTheme(newTheme) {
+  // eslint-disable-next-line no-console
+  console.log(`CHANGE THEME - ${newTheme}`);
+
+  const themeElement = document.querySelector("#theme");
+
+  if (themeElement) {
+    themeElement.remove();
+  }
+
+  if (themes[newTheme]) {
+    // eslint-disable-next-line no-console
+    console.log(`THEME ALREADY LOADED - ${newTheme}`);
+
+    document.head.appendChild(themes[newTheme]);
+
+    return;
+  }
+
+  if (newTheme === "dark") {
+    // eslint-disable-next-line no-console
+    console.log(`LOADING THEME - ${newTheme}`);
+
+    import(/* webpackChunkName: "dark" */ "./style.scss?dark").then(() => {
```

Comment:
Is the chunk name comment required here, or is it optional? If it's required, would there be  any way to define it in the webpack config rather than the source?

Just thinking out loud: It might be possible to use a pitching loader to expand a static `./style.scss` import to one per theme (i.e. `../style.scss?dark`, `./style.scss?light`, `./style.scss?n...`), and then use matching resource queries.

Reply:
This is for tests and examples, for production I recommend to not use comments, because you will have less names for files, so you will send less bytes for client

Reply:
Got it, thank you. Would it make sense to revise this example to use production best practices?

Reply:
Feel free to send a PR and fix it

Reply:
I can certainly fix the `Self` references in the README. Stand by for that.

However, I don't actually know how to replace `webpackChunkName` comments with something more first class (for example. something we'd see in `webpack.config.js`). This is one of the reasons I've been following this issue so closely.